language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -24,13 +24,17 @@ main = do
     damlc <- locateRunfiles (mainWorkspace </> "compiler" </> "damlc" </> exe "damlc")
     daml2ts <- locateRunfiles (mainWorkspace </> "language-support" </> "ts" </> "codegen" </> exe "daml2ts")
     davl <- locateRunfiles ("davl" </> "released")
-    defaultMain $ tests damlc daml2ts davl
+    yarnPath : args <- getArgs
+    yarn <- locateRunfiles (mainWorkspace </> yarnPath)
+    withTempDir $ \rootDir -> do
```

Comment:
Why not use one temp dir per test? seems easier to manage and avoids accidentally ending up with colliding directories

Reply:
It was phrased earlier as one temp dir per test but it's like this now to cache yarn dependencies by sticking everything under a common root.

Reply:
Makes sense, maybe add a comment?

Reply:
You could also move the creation of the tempdir in the tasty resource as I suggested below.